### ISSUE TITLE
implement a lightweight Matter over WiFi protocol stack in ESPHome

### ISSUE THREAD

**Issue by lboue** [esphome/feature-requests#2722]
**Describe the problem you have/What new integration you would like**

Would it be possible to implement a lightweight Matter over WiFi protocol stack in ESPHome?

**Please describe your use case for this integration and alternatives you've tried:**

The aim is to be able to control these devices directly via a Matter controller.

**Additional context**

**Comment by Hedda** [comment]
Duplicate of https://github.com/esphome/feature-requests/issues/1430

**Comment by lboue** [comment]
> Duplicate of #1430 as that applies to Matter over IP on both Wi-Fi and Ethernet

This issue #1430 was opened in 2021 and is becoming a catch-all.

**Comment by Hedda** [comment]
> This issue #1430 was opened in 2021 and is becoming a catch-all. There is a discussion about Zigbee, Thread Border Router, etc.

I disagree, I think https://github.com/esphome/feature-requests/issues/1430 is clearly only about connectivity using the high-level Matter standrard (and not really about Thread).

You might be thinking about https://github.com/esphome/feature-requests/issues/1397 which is a discussion about OpenThread and Zigbee on ESP32 (like ESP32-H2/ESP32-C6)

**Comment by lboue** [comment]
> > This issue #1430 was opened in 2021 and is becoming a catch-all. There is a discussion about Zigbee, Thread Border Router, etc.
> 
> I disagree.
> 
> #1430 is only about connectivity using the Matter application-layer, (it is not really about media-layer, e.g. Thread/OpenThread).


Border Router
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1010939789
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1268384267
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1259198292

OpenThread
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1268384267
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1259198292
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1435639413


Zigbee
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1010939789
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1259284389
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1268384267
https://github.com/esphome/feature-requests/issues/1430#issuecomment-1435639413

> Are you thinking about #1397? That is a discussion about OpenThread media-layer (and Zigbee) on ESP32 (like ESP32-H2)
No

**Comment by nicx** [comment]
waiting for matter support too... is this somehow in development?

**Comment by mhalano** [comment]
I think we need to finish Thread support first. I know that can be a WI-Fi-only implementation, but that's not the standard.

**Comment by nicx** [comment]
@mhalano why should Wi-Fi not be the standard? Both protocols are completly standardized with Matter. In addition to these 2 wireless protocols also Ethernet is included in the matter standard.

**Comment by Hedda** [comment]
Again this is another reason why this is a duplicate of this other request:

- https://github.com/esphome/feature-requests/issues/1430

Anyway, Matter of WiFi follows the standard, Matter for Ethernet follows the standard, and Matter over Thread follows the standard, there is nothing in the standards that you need to support more than one of those transport protocols.

**Comment by mhalano** [comment]
What I mean, is: ESPHome is not a finished product, like my smartplug that just supports Wi-Fi, but it's a platform. We need to implement Matter with full support because someone want to use Wi-Fi, but other folk will want to use Thread, and a third one will want Ethernet. Said that, I also think we could have a preview implementation just with Wi-Fi, that, I think, is the most used system. But to consider the solution stable and ready, we need all three implemented. Again, ESPHome isn't a finished product, but a platform.

**Comment by Hedda** [comment]
Hence why this is a duplicate of https://github.com/esphome/feature-requests/issues/1430

**Comment by goloveychuk** [comment]
Fyi thread is implemented in esphome

**Comment by Hedda** [comment]
Thread or Wi-Fi does not matter (yes a pun), the Matter application layer is independent of the transport protocol, so this is still a duplicate of https://github.com/esphome/feature-requests/issues/1430


Again, the transport layer, network layer, and physical layer are totally separated from the application layer in the Matter stack:

<img width="1300" height="629" alt="Image" src="https://github.com/user-attachments/assets/479b5a77-98bf-4f25-a2ee-4d9df45533c0" />

The Matter application layer sits above the other layers of the protocol stack, relying on them for reliable communication. 

- Transport Layer: The application layer interacts with the transport layer to send and receive messages over an IP network. 

- Connectivity: It can use various network connectivity technologies, such as Wi-Fi, Thread, and Ethernet, to establish communication paths with other devices. 

- Security: A security layer ensures that data is authenticated and encrypted before being sent to the transport layer for delivery to the target device.
